lib/spack/spack/database.py

```diff
@@ -65,8 +66,13 @@
     (Version('0.9.3'), Version('5')),
 ]
 
-# Timeout for spack database locks in seconds
-_db_lock_timeout = 120
+# Default timeout for spack database locks in seconds, which we want to provide
+# a fairly quick turnaround for parallel installs
+_db_lock_timeout = 3
```

Comment:
Do we need a quick turnaround on failures for the database itself?  I'm not clear on why it's not just the prefix locks that need a fast timeout.

If a process fails to update the DB, does it retry and move on?  I think it may be fairly easy for the DB not to get updated after a successful install if the timeout is this low, if the filesystem is busy.  Or, actually, even if the DB gets to be a certain size, it may take more than 3 seconds to write, in which case any concurrent lock attempts will fail.

Reply:
Fair point.  What is a more reasonable setting?

At one point I was having batch jobs terminate with little to no progress that appeared to be tied to the database lock timeout.  I did not see the problem after reducing the timeout.

Reply:
I expanded on the comments for the timeouts to include `None` and explain some of their uses and trade-offs.

Reply:
I thought it was causing (at one point) some of my batch jobs to terminate without making progress and I saw a significant improvement by reducing it.

I will double-check to see if that is still the case though.